[Synthetics] Add monitor interval value in synthetics data !!

shahzad31 opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/229466

Adds the monitor interval value (in seconds) to the synthetics data sent to Heartbeat, enabling downstream consumers to know the configured check frequency directly from the monitor data.

Changes:

- Added 'monitor.interval' field (t.number, in seconds) to HeartbeatFieldsCodec
- Added periodToSeconds() utility (reuses periodToMs)
- Included 'monitor.interval' in formatHeartbeatRequest (public locations) and processorsFormatter (private/Fleet locations)
- Updated all related unit and integration test fixtures

<!--ONMERGE {"backportTargets":["9.4"]} ONMERGE-->

shahzad31 committed:
interval

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/323663)
* Commit: cf29e9a302b8927d05729c706252ea9925771076

### Failed CI Steps
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985028-7c2b-4e27-826b-e1215265304e)
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985040-9d72-4fca-82a3-311f5e9f3044)
* [FTR Configs #118](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985028-7c69-4b8d-be61-f56cc198951d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985028-7c2b-4e27-826b-e1215265304e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/323663/jobs/01985028-7c2b-4e27-826b-e1215265304e/artifacts/01985040-8474-4381-8daa-8b1a94f33c0e) FTR Configs #<span></span>57 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests PrivateLocationCreateMonitor added an integration for previously added monitor
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985040-9d72-4fca-82a3-311f5e9f3044) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/323663/jobs/01985040-9d72-4fca-82a3-311f5e9f3044/artifacts/0198504b-363c-4a77-b8df-4618b752e689) FTR Configs #<span></span>57 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests PrivateLocationCreateMonitor added an integration for previously added monitor
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/323663#01985028-7c69-4b8d-be61-f56cc198951d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/323663/jobs/01985028-7c69-4b8d-be61-f56cc198951d/artifacts/01985042-01f1-4b02-9b1d-0a200ccb13aa) FTR Configs #<span></span>118 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests PrivateLocationCreateMonitor added an integration for previously added monitor

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323663","number":323663,"commit":"cf29e9a302b8927d05729c706252ea9925771076"}],"number":323663}
buildkite-pr-comment-->

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into interval-value

shahzad31 committed:
Update tests to include monitor.interval field in expected output

Made-with: Cursor

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into interval-value

benakansara reviewed (approved):
LGTM

shahzad31 committed:
update tests

shahzad31 committed:
Merge branch 'main' into interval-value

shahzad31 committed:
Merge branch 'main' into interval-value

shahzad31 committed:
Add monitor interval fields to synthetics inspect monitor tests

- Added 'monitor.interval' field with values 300 and 600 to the test configuration for synthetics inspect monitor.
- This change enhances the test setup by specifying monitor intervals for different scenarios, improving test accuracy and coverage.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/417467)
* Commit: 4d3f33b2ce449a1dd7321ab911bf0b635437187a

### Failed CI Steps
* [FTR Configs #130](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2bf6-00c6-4329-bb06-3527cbdd19b2)
* [FTR Configs #130](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2c12-349e-4dac-872f-695171546154)
* [FTR Configs #147](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2bf6-00d2-4b00-be31-e58a7bcc8ef4)
* [FTR Configs #147](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2c0f-7887-4769-af70-56b4542376fc)
* [FTR Configs #214](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2bf6-010a-4b8c-8021-58bcd76c555e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2bf6-00d2-4b00-be31-e58a7bcc8ef4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/417467/jobs/019d2bf6-00d2-4b00-be31-e58a7bcc8ef4/artifacts/019d2c0f-60ea-4704-8941-17420a5a98cf) FTR Configs #<span></span>147 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests inspectSyntheticsMonitor inspect http monitor in private location
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2c0f-7887-4769-af70-56b4542376fc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/417467/jobs/019d2c0f-7887-4769-af70-56b4542376fc/artifacts/019d2c25-c79b-40de-8aca-9d59c1b6323f) FTR Configs #<span></span>147 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests inspectSyntheticsMonitor inspect http monitor in private location
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2bf6-00c6-4329-bb06-3527cbdd19b2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/417467/jobs/019d2bf6-00c6-4329-bb06-3527cbdd19b2/artifacts/019d2c12-1eaf-4f35-a4a4-3a3c4b23fd48) FTR Configs #<span></span>130 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests inspectSyntheticsMonitor inspect http monitor in private location
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/417467#019d2c12-349e-4dac-872f-695171546154) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/417467/jobs/019d2c12-349e-4dac-872f-695171546154/artifacts/019d2c27-e1e1-419f-ab18-b6f1165af2b8) FTR Configs #<span></span>130 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests inspectSyntheticsMonitor inspect http monitor in private location

### History
* :broken_heart: [Build #417078](https://buildkite.com/elastic/kibana-pull-request/builds/417078) failed f55df2f0023374de9d015784f4643311b16710e0
* :broken_heart: [Build #416079](https://buildkite.com/elastic/kibana-pull-request/builds/416079) failed 0ffc75f9f57b499b60212f9500b4d48de58bfbba
* :broken_heart: [Build #323663](https://buildkite.com/elastic/kibana-pull-request/builds/323663) failed cf29e9a302b8927d05729c706252ea9925771076

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/417467","number":417467,"commit":"4d3f33b2ce449a1dd7321ab911bf0b635437187a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/417078","number":417078,"commit":"f55df2f0023374de9d015784f4643311b16710e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/416079","number":416079,"commit":"0ffc75f9f57b499b60212f9500b4d48de58bfbba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323663","number":323663,"commit":"cf29e9a302b8927d05729c706252ea9925771076"}],"number":417467}
buildkite-pr-comment-->

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into interval-value

shahzad31 committed:
update tests

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/426615)
* Commit: 0543d1d8a89b76110de5e632e0f0c93a57427b77

### Failed CI Steps
* [FTR Configs #183](https://buildkite.com/elastic/kibana-pull-request/builds/426615#019d8707-5326-4a3f-a17d-ea3d706119c7)
* [Scout: [ security / entity_store ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/426615#019d8711-ce27-4e3d-8082-54359b33397e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/426615#019d8711-ce27-4e3d-8082-54359b33397e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/426615/jobs/019d8711-ce27-4e3d-8082-54359b33397e/artifacts/019d8731-4a8e-4325-b4e4-67dd0042c5a6) Scout: [ security / entity_store ] plugin / local-stateful-classic - Entity Store History Snapshot - history snapshot: copies latest to history index and resets behaviors on latest

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2e5f8d4d17c4b6504912acd66baea8f985deab0c) | [after](https://github.com/elastic/kibana/commit/0543d1d8a89b76110de5e632e0f0c93a57427b77) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.1MB | 1.1MB | +28.0B |



### History
* :broken_heart: [Build #417467](https://buildkite.com/elastic/kibana-pull-request/builds/417467) failed 4d3f33b2ce449a1dd7321ab911bf0b635437187a
* :broken_heart: [Build #417078](https://buildkite.com/elastic/kibana-pull-request/builds/417078) failed f55df2f0023374de9d015784f4643311b16710e0
* :broken_heart: [Build #416079](https://buildkite.com/elastic/kibana-pull-request/builds/416079) failed 0ffc75f9f57b499b60212f9500b4d48de58bfbba
* :broken_heart: [Build #323663](https://buildkite.com/elastic/kibana-pull-request/builds/323663) failed cf29e9a302b8927d05729c706252ea9925771076

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/426615","number":426615,"commit":"0543d1d8a89b76110de5e632e0f0c93a57427b77"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/417467","number":417467,"commit":"4d3f33b2ce449a1dd7321ab911bf0b635437187a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/417078","number":417078,"commit":"f55df2f0023374de9d015784f4643311b16710e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/416079","number":416079,"commit":"0ffc75f9f57b499b60212f9500b4d48de58bfbba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323663","number":323663,"commit":"cf29e9a302b8927d05729c706252ea9925771076"}],"number":426615}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 9.4

https://github.com/elastic/kibana/actions/runs/24504829275

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.4|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/263723">](https://github.com/elastic/kibana/pull/263723)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
